clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7790,6 +7790,21 @@ bool Sema::CheckIntelSYCLPtrAnnotationBuiltinFunctionCall(unsigned BuiltinID,
   return false;
 }
 
+static llvm::APSInt getSYCLAllocaDefaultSize(const ASTContext &Ctx,
+                                             const VarDecl *VD) {
+  assert(VD && "Expecting valid declaration");
+  APValue *SpecializationId = VD->evaluateValue();
+  assert(SpecializationId && "Expecting a non-null SpecializationId");
+  assert(SpecializationId->getKind() == APValue::ValueKind::Struct &&
+         "Expecting SpecializationId to be of kind Struct");
+  assert(SpecializationId->getStructNumFields() == 1 &&
+         "Expecting SpecializationId to have a single field for the default "
+         "value");
+  APValue Default = SpecializationId->getStructField(0);
+  assert(Default.isInt() && "Expecting the default value to be an integer");
```

Comment:
Nit: Same here. I think getInt() has an assert. I don't insist on removing this though since the error message is nicer here.